invokeai/app/invocations/facetools.py

```diff
@@ -0,0 +1,654 @@
+import math
+
+import cv2
+import mediapipe as mp
+import numpy as np
+from PIL import Image, ImageDraw, ImageFilter, ImageOps
+
+from invokeai.app.invocations.baseinvocation import (
+    BaseInvocation,
+    BaseInvocationOutput,
+    InputField,
+    InvocationContext,
+    OutputField,
+    invocation,
+    invocation_output,
+)
+from invokeai.app.invocations.primitives import ImageField, ImageOutput
+from invokeai.app.models.image import ImageCategory, ResourceOrigin
+
+
+@invocation_output("face_mask_output")
+class FaceMaskOutput(BaseInvocationOutput):
+    """Base class for FaceMask output"""
+
+    image: ImageField = OutputField(description="The output image")
+    width: int = OutputField(description="The width of the image in pixels")
+    height: int = OutputField(description="The height of the image in pixels")
+    mask: ImageField = OutputField(description="The output mask")
+
+
+@invocation_output("face_off_output")
+class FaceOffOutput(BaseInvocationOutput):
+    """Base class for FaceOff Output"""
+
+    bounded_image: ImageField = OutputField(default=None, description="Original image bound, cropped, and resized")
```

Comment:
Is this an optional output field? I mean does it sometimes output `None`? If so, it should be typed like `bounded_image: Optional[ImageField]`. If it's always going to exist, remove the default.